configuration/config2/SchemaItem.hpp

```diff
@@ -0,0 +1,112 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2017 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+#ifndef _CONFIG2_CONFIGITEM_HPP_
+#define _CONFIG2_CONFIGITEM_HPP_
+
+#include <string>
+#include <memory>
+#include <set>
+#include <map>
+#include "SchemaType.hpp"
+#include "SchemaValue.hpp"
+
+
+class SchemaItem : public std::enable_shared_from_this<SchemaItem>
+{
+    public:
+
+        SchemaItem(const std::string &name, const std::string &className = "category", const std::shared_ptr<SchemaItem> &pParent = nullptr);
+        //ConfigItem(const ConfigItem &ci);
```

Comment:
I was hold this code back because I thought I might need to activate it in the future (a copy constructor). But I decided to remove it. I can always rewrite it if needed.